### ISSUE TITLE
LocalJumpError when running simple script

### ISSUE THREAD

**Issue by bjorne** [iconara/puck#17]
A `LocalJumpError` is raised when running a script. I see this error for both JRuby 1.7.19 and 9.0.0.0. It looks like e537f0f9f8149a1b308936ed41c4dc3a4a0c395e is to blame. I suspect we have not seen this before because most processes are long running, and if the script explicitly exits the return statement will never be reached.

```
$ java -jar build/foo.jar foo
Hello
LocalJumpError: unexpected return
  block in classpath:jar-bootstrap.rb at classpath:jar-bootstrap.rb:11
                                 each at org/jruby/RubyArray.java:1560
                                <top> at classpath:jar-bootstrap.rb:6
$ cat bin/foo
#!/usr/bin/env ruby
puts 'Hello'
```

**Comment by bjorne** [comment]
I think I must have mistaken myself with regards to JRuby 1.7.19. Only Ruby 2.0+ interpreters seem to be affected. I suppose it is no longer legal to return from blocks (without an enclosing method) in Ruby 2.0+?

```
$ for i in $(rvm list | ggrep -Eo 'j?ruby-(2.|1.7.|9.)[^ ]*'); do echo $i; rvm $i do ruby -e 'begin; [1].each { |x| return }; rescue; puts "fail"; end'; done
jruby-1.7.10
jruby-1.7.11
jruby-1.7.12
jruby-1.7.13
jruby-1.7.14
jruby-1.7.16
jruby-1.7.16.1
jruby-1.7.18
jruby-1.7.19
jruby-1.7.2
jruby-1.7.4
jruby-1.7.6
jruby-1.7.9
jruby-9.0.0.0
fail
ruby-2.0.0-p353
fail
ruby-2.0.0-p481
fail
ruby-2.1.4
fail
ruby-2.2.0
fail
ruby-2.2.1
fail
```

**Comment by iconara** [comment]
This was fixed in v1.2.0
